### PR TITLE
Validate User API responses in stage and test

### PR DESCRIPTION
**Pull request by fridex**
## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/user-api/issues/1253
Related: https://github.com/thoth-station/user-api/pull/1254
Release: https://github.com/thoth-station/user-api/pull/1258

## This introduces a breaking change

- [x] No

### CODE REVIEW

File: user-api/base/configmap.yaml [hunk 1]
```diff
@@ -6,3 +6,4 @@ metadata:
 data:
   api-https: "1"
   amun-api-url: "http://amun-api-thoth-test-core.cloud.paas.psi.redhat.com/api/v1"
+  validate-responses: "0"
```
Comment (fridex) [inline user-api/base/configmap.yaml:9]: A merge conflict with https://github.com/thoth-station/thoth-application/pull/913, will rebase on approval.

File: user-api/base/deploymentconfig.yaml [hunk 2]
```diff
@@ -36,6 +36,11 @@ spec:
               value: "khemenu.thoth-station.ninja"
             - name: "THOTH_SENTRY_IGNORE_EXCEPTION"
               value: "thoth.user_api.exceptions.NotFoundException,thoth.storages.exceptions.NotFoundError"
+            - name: THOTH_API_VALIDATE_RESPONSES
+              valueFrom:
+                configMapKeyRef:
+                  key: management-api
+                  name: thoth
```
Comment (harshad16) [inline user-api/base/deploymentconfig.yaml:43]:
```suggestion
                  key: validate-responses
                  name: user-api
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thanks for the addition.
A minor change is required based on the reference.

File: user-api/base/deploymentconfig.yaml (hunk 2 above)
Comment (fridex) [reply user-api/base/deploymentconfig.yaml]: Nice catch, thanks! :+1:

**Review by harshad16 — COMMENTED** [review]
/approve
This can go in.
thanks for the addition.

**Pull request merged**